react/fields/TextField/TextField.js

```diff
@@ -163,64 +167,58 @@ export default class TextField extends Component {
     );
   }
 
-  renderHelp() {
-    const { message, help } = this.props;
-
-    if (message || !help) {
-      return;
-    }
-
-    const { helpProps } = this.props;
-    const allHelpProps = combineClassNames(helpProps, styles.help);
-
-    return (
-      <p {...allHelpProps}>
-        {help}
-      </p>
-    );
-  }
-
   renderMessage() {
-    const { message } = this.props;
+    const { message, valid } = this.props;
 
     if (!message) {
       return;
     }
 
-    const { messageProps } = this.props;
-    const allMessageProps = combineClassNames(messageProps, styles.message);
+    const { critical, positive, secondary, ...restMessageProps } = this.props.messageProps;
+    const allMessageProps = combineClassNames(restMessageProps, styles.message);
 
     return (
-      <p {...allMessageProps}>
-        {this.renderIcon()}
+      <Text {...allMessageProps} critical={(valid === false && !secondary) || critical} positive={(valid === true && !secondary) || positive} secondary={typeof valid === 'undefined' || secondary}>
```

Comment:
Can i suggest from a legibility perspective putting these props on indented new lines? Just makes it a lot easier to scan, sorry minor i know